### DISCUSSION TITLE
Clustering based on note name

### DISCUSSION THREAD

**Discussion by joesanga** [SkepticMystic/graph-analysis#72] [General]
Hi there, 

Is there a way to cluster notes (scattered throughout the Obsidian vault), based only on their title? Ignoring the links, backlinks, content inside the note etc., I simply would like to cluster notes based on the title name. This can help avoid duplication. Any thoughts please? This is WITHOUT me having to enter a particular keyword. I'd like the plugin to generate title-based clusters please. Thank you.

**Comment by hittjwX** [comment]
Tell me a little more about your use case. 

The following will pull exact match titles using `dataview`:

```dataview
TABLE rows.file.link
GROUP BY file.name
WHERE length(rows) > 1
```

It is possible to have duplicate filenames from imports. They will cause problems in Obsidian around wiki links. Obsdian won't let you create duplicate filenames in its editor.

There are times when filenames are similar, but the content is different. For example, `202601141615 Successful Membership Site` and `202601141623 Successful Membership Site` both represent work in a subject area.

The Zettelkasten serial tells me they are different. A script like [find_similar_filenames.py](https://github.com/jwhco/scripts/blob/main/TidyObsidian/find_similar_filenames.py) will consider `2024-11-17.md` and `2025-02-22.md` to be similar. The two in the example above would also be considered similar.

This block is also considered similar:

```
=== Similar group: n.jpg ===
  assets/458285736_822202303394743_6194764532129146824_n.jpg
  assets/428611518_925400872611198_4923171366014563786_n.jpg
  assets/461497419_1057998406110093_236679185093860911_n.jpg
```

Jaccard and Co-Citation rely on broader contexts to consider relatedness. Even when it does with counts and references. Filenames don't offer that much context.

Placing an algorithm like the referenced script in a sidebar would introduce significant noise. Where similar filenames are helpful is to find exact matches across directories. 

```
=== Similar group: value of free information.md ===
  Queue/Aweber/Aweber-iun-ck1-d0830i/2900-2010_07_26_0723 Value of Free Information.md
  Queue/OneNote/ISR/Editorial/S1-DRAFT/107_2010_07_26_0723_Value_of_free_information.md
  Queue/OneNote/ISR/Editorial/S1-DRAFT/106_2010_07_26_0723_Value_of_free_information.md
```

To find an exact match in Obsidian, try the Dataview. To find similar, you can type a phrase in the finder (Ctrl-O), then select from the list. Otherwise, tell me more about the use case.

**Comment by hittjwX** [comment]
Did this answer your question? You can also use a Zettelkasten-style key at the beginning of all filenames.
